internal/eventstream/eventstream.go

```diff
@@ -0,0 +1,70 @@
+package eventstream
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/alecthomas/types/pubsub"
+)
+
+// EventStream is a stream of events that can be published and subscribed to, that update a materialized view
+type EventStream[T View] interface {
+	Publish(event Event[T]) error
+
+	View() T
+
+	Subscribe() <-chan Event[T]
```

Comment:
This should document that it will only receive events published after subscribing.

Reply:
This is fundamentally kinda racy, as there is no way to subscribe and get the current view in an atomic operation. I don't think this is important at this stage though.